flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/enumerator/IncrementalSourceEnumerator.java

```diff
@@ -74,11 +85,21 @@ public IncrementalSourceEnumerator(
         this.sourceConfig = sourceConfig;
         this.splitAssigner = splitAssigner;
         this.readersAwaitingSplit = new TreeSet<>();
+
+        // when restored from state, if the split assigner is assigning snapshot
+        // splits or has already assigned all splits, send wakeup event to
+        // SourceReader, SourceReader can omit the event based on its own status.
+        if (isAssigning(splitAssigner.getAssignerStatus())
+                || isAssigningFinished(splitAssigner.getAssignerStatus())) {
+            streamReaderIsSuspended = true;
+        }
     }
 
     @Override
     public void start() {
         splitAssigner.open();
+        suspendBinlogReaderIfNeed();
+        wakeupBinlogReaderIfNeed();
```

Comment:
Can we name it abstractly such as `suspendStreamReaderIfNeed` and `wakeupStreamReaderIfNeed `?

Reply:
Addressed in https://github.com/ververica/flink-cdc-connectors/pull/1838/commits/7490d9b9785adb1db065b68cd2305458e46920aa